src/main/java/cn/nukkit/Player.java

```diff
@@ -2016,7 +2016,7 @@ protected void processLogin() {
 
         this.adventureSettings = new AdventureSettings(this)
                 .set(Type.WORLD_IMMUTABLE, isAdventure() || isSpectator())
-                .set(Type.WORLD_BUILDER, !isAdventure() && !isSpectator())
+                .set(Type.WORLD_BUILDER, !isAdventure() || !isSpectator())
```

Comment:
That would be always true

GameMode | `not isAdventure` | `not isSpectator` | `(not isAdventure) OR (not isSpectator)`
---------|----|-----|--------
survival | true | true | true
creative | true | true | true
adventure | false | true | true
spectator | true | false | true

Reply:
I don't know why but on my server it solved the spectator problem in spc and adventure gamemode and I could still break the blocks in creative and survival.

Reply:
GameMode | not isAdventure | not isSpectator | (not isAdventure) OR (not isSpectator)
-- | -- | -- | --
survival | true | true | true
creative | true | true | true
adventure | false | true | false
spectator | true | false | false

you are wrong 🤔🤔